RoomBookingSystem/app/Http/Controllers/BookingRequestController.php

```diff
@@ -36,12 +38,58 @@ public function Create(Request $request)
         return response()->json(['status' => 'ok'], 200);
     }
 
-    public function myReceivedRequests(Request $request)
+    public function approve(Request $request, BookingRequest $bookingRequest)
+    {
+        $user = auth()->user();
+    
+        if ($user->id !== $bookingRequest->user_id && $user->role !== 'admin') {
+            return response()->json(['message' => 'Forbidden'], 403);
+        }
```

Comment:
Wouldn't it make sense to remove the role check, and only have the user id check?
```suggestion
        if ($user->id !== $bookingRequest->user_id) {
            return response()->json(['message' => 'Forbidden'], 403);
        }
```

Reply:
kommer an på om admins skulle kunne acceptere alle?